crates/bevy_ecs/src/world/mod.rs

```diff
@@ -746,15 +739,61 @@ impl World {
         ticks.is_changed(self.last_change_tick(), self.read_change_tick())
     }
 
-    /// Gets a mutable reference to the resource of the given type, if it exists. Otherwise returns
+    /// Gets a reference to the resource of the given type
+    ///
+    /// # Panics
+    ///
+    /// Panics if the resource does not exist.
+    /// Use [`get_resource`](World::get_resource) instead if you want to handle this case.
+    ///
+    /// If you want to instead insert a value if the resource does not exist,
+    /// use [`get_resource_or_insert_with`](World::get_resource_or_insert_with).
+    #[inline]
+    pub fn resource<R: Resource>(&self) -> &R {
+        self.get_resource().unwrap_or_else(|| {
+            panic!(
+                "Requested resource {} does not exist. Did you forget to add it to the `World`, call `app.add_event` or add a plugin that contains it?",
```

Comment:
Small nitpick and question from a non native speaker
```
Did you forget to add it to the `World`, call `app.add_event` or add a plugin that contains it?
```
The part after the comma doesn't feel right as part of a question. I would have gone with.
```
Did you forget to add it to the `World`? Make sure you call `app.add_event` or add a plugin that contains it.
```
That said, if it feels right for you native speakers, then just ignore this comment 🙂

Reply:
Yeah, I think I can make this clearer :)